texar/torch/modules/connectors/connectors.py

```diff
@@ -69,80 +69,74 @@ def _assert_same_size(outputs: TensorStruct,
                 "The output size does not match the required output_size")
 
 
-def _get_tensor_depth(x: torch.Tensor) -> int:
-    r"""Returns the size of a tensor excluding the first dimension
-    (typically the batch dimension).
+def _get_sizes(sizes: List[Any]) -> List[int]:
+    r"""
 
     Args:
-        x: A tensor.
+        sizes: A list of ``int`` or ``torch.Size``. If each element is of type
+            ``torch.Size``, the size is computed by taking the product of the
+            shape.
+
+    Returns:
+        A list of sizes with ``torch.Size`` replaced by product of its
+        individual dimensions
     """
-    return int(np.prod(x.size()[1:]))
+    if isinstance(sizes[0], torch.Size):
+        size_list = [np.prod(shape) for shape in sizes]
+    else:
+        size_list = sizes
+
+    return size_list
 
 
 def _sum_output_size(output_size: OutputSize) -> int:
     r"""Return sum of all dim values in :attr:`output_size`
 
     Args:
-        output_size: Can be an ``Integer``, a ``torch.Size``, or a (nested)
-            ``tuple`` of ``Integers`` or ``torch.Size``.
+        output_size: Can be an ``int``, a ``torch.Size``, or a (nested)
+            tuple of ``int`` or ``torch.Size``.
     """
     flat_output_size = nest.flatten(output_size)
-
-    if isinstance(flat_output_size[0], torch.Size):
-        size_list = [0] * len(flat_output_size)
-        for (i, shape) in enumerate(flat_output_size):
-            size_list[i] = np.prod([dim for dim in shape])
-    else:
-        size_list = flat_output_size
+    size_list = _get_sizes(flat_output_size)
     ret = sum(size_list)
     return ret
 
 
 def _mlp_transform(inputs: TensorStruct,
                    output_size: OutputSize,
                    linear_layer: Optional[LinearLayer] = None,
-                   activation_fn: Optional[ActivationFn] = None,
-                   ) -> Any:
+                   activation_fn: Optional[ActivationFn] = None) -> Any:
     r"""Transforms inputs through a fully-connected layer that creates
     the output with specified size.
 
     Args:
-        inputs: A ``Tensor`` of shape ``[batch_size, ..., finale_state]``
+        inputs: A tensor of shape ``[batch_size, d1, ..., dn]``
             (i.e., batch-major), or a (nested) tuple of such elements.
-            A Tensor or a (nested) tuple of Tensors with shape
+            A tensor or a (nested) tuple of tensors with shape
             ``[max_time, batch_size, ...]`` (i.e., time-major) can
```

Comment:
Is this PR include all changes in https://github.com/asyml/texar-pytorch/pull/162? It looks not to me. E.g., the correct doc of `inputs` here should be: https://github.com/asyml/texar-pytorch/pull/162/files#diff-4a3f72feda3ae73445008a08ee31878aR99

Reply:
Added this sentence - `The dimensions ``d1, ..., dn`` will be flattened and transformed by a dense layer.` I retain

```
inputs: A tensor of shape ``[batch_size, d1, ..., dn]``
            (i.e., batch-major), or a (nested) tuple of such elements.
            A tensor or a (nested) tuple of tensors with shape
            ``[max_time, batch_size, ...]`` (i.e., time-major) can
            be transposed to batch-major using
            :func:`~texar.torch.utils.transpose_batch_time` prior to this
            function.
```

in this PR (which was deleted in #162) as I think this is more informative. #162 requested for 3 changes - one was this doc change, other two requested to add `_get_sizes` method in this file.

Reply:
The original doc `A tensor or a (nested) tuple of tensors with shape [max_time, batch_size, ...] (i.e., time-major) ...` is incorrect. Let's copy the new docs from https://github.com/asyml/texar-pytorch/pull/162 directly. 

It corresponds to TF: https://github.com/asyml/texar/blob/master/texar/tf/modules/connectors/connectors.py#L78

Reply:
Fixed in 570c7ab